test/garbage_collection_test.js

```diff
@@ -26,7 +26,12 @@ suite('garbage collection tests', function () {
       delayFactor: 15 * 1000,
       randomizationFactor: 0.25
     },
-    log: logger()
+    log: logger(),
+    stats: {
+      timeGen: async function (series, fn) {
+        return await fn;
```

Comment:
Would it be helpful to enforce that `fn` is a Promise (not a function..) here?  If it's not a Promise, then the timing will be 0.